src/ring_theory/noetherian.lean

```diff
@@ -391,6 +394,12 @@ theorem set_has_maximal_iff_noetherian {R M} [ring R] [add_comm_group M] [module
   (∀ a : set $ submodule R M, a.nonempty → ∃ M' ∈ a, ∀ I ∈ a, M' ≤ I → I = M') ↔ is_noetherian R M :=
 by rw [is_noetherian_iff_well_founded, well_founded.well_founded_iff_has_max']
 
+/-- If `∀ I > J, P I` implies `P J`, then `P` holds for all submodules. -/
+lemma proposition_noetherian_induction {R M} [ring R] [add_comm_group M] [module R M] [is_noetherian R M]
+  {P : submodule R M → Sort*} (hgt : ∀ I, (∀ J > I, P J) → P I)
+  (I : submodule R M) : P I :=
```

Comment:
I still got a linting error after the changes. Looks like `well_founded.recursion` is defined kind of weird so we can only return a `Prop` instead of all possible `Sort`s.
```suggestion
lemma is_noetherian.induction {R M} [ring R] [add_comm_group M] [module R M] [is_noetherian R M]
  {P : submodule R M → Prop} (hgt : ∀ I, (∀ J > I, P J) → P I)
  (I : submodule R M) : P I :=
```